Bottom: be69ee106f6885ceef786f3070dcc0a9796cc1dc
Top:    b56790617c0ea52f1d0d748d0fbfa6ef95d0a641
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-12-22 17:17:57 +0800

Refresh of tuntap-rename-struct

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 6ab9e4e..d29fa81 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2052,7 +2052,7 @@ static int __tun_set_ebpf(struct tun_struct *tun, struct tun_prog **prog_p,
 	spin_lock_bh(&tun->lock);
 	old = rcu_dereference_protected(*prog_p,
 					lockdep_is_held(&tun->lock));
-	rcu_assign_pointer(tun->steering_prog, new);
+	rcu_assign_pointer(*prog_p, new);
 	spin_unlock_bh(&tun->lock);
 
 	if (old)
